source/Personality.cpp

```diff
@@ -20,44 +20,19 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include "DataWriter.h"
 
 #include <map>
+#include <vector>
 
 using namespace std;
 
 namespace {
-	const int PACIFIST = (1 << 0);
-	const int FORBEARING = (1 << 1);
-	const int TIMID = (1 << 2);
-	const int DISABLES = (1 << 3);
-	const int PLUNDERS = (1 << 4);
-	const int HUNTING = (1 << 5);
-	const int STAYING = (1 << 6);
-	const int ENTERING = (1 << 7);
-	const int NEMESIS = (1 << 8);
-	const int SURVEILLANCE = (1 << 9);
-	const int UNINTERESTED = (1 << 10);
-	const int WAITING = (1 << 11);
-	const int DERELICT = (1 << 12);
-	const int FLEEING = (1 << 13);
-	const int ESCORT = (1 << 14);
-	const int FRUGAL = (1 << 15);
-	const int COWARD = (1 << 16);
-	const int VINDICTIVE = (1 << 17);
-	const int SWARMING = (1 << 18);
-	const int UNCONSTRAINED = (1 << 19);
-	const int MINING = (1 << 20);
-	const int HARVESTS = (1 << 21);
-	const int APPEASING = (1 << 22);
-	const int MUTE = (1 << 23);
-	const int OPPORTUNISTIC = (1 << 24);
-	const int MERCIFUL = (1 << 25);
-	const int TARGET = (1 << 26);
-	const int MARKED = (1 << 27);
-	const int LAUNCHING = (1 << 28);
-	const int DARING = (1 << 29);
-	const int SECRETIVE = (1 << 30);
-	const int RAMMING = (1 << 31);
-
-	const map<string, int> TOKEN = {
+	enum PersonalityTrait {
+		PACIFIST, FORBEARING, TIMID, DISABLES, PLUNDERS, HUNTING, STAYING, ENTERING, NEMESIS,
+		SURVEILLANCE, UNINTERESTED, WAITING, DERELICT, FLEEING, ESCORT, FRUGAL, COWARD, VINDICTIVE,
+		SWARMING, UNCONSTRAINED, MINING, HARVESTS, APPEASING, MUTE, OPPORTUNISTIC, MERCIFUL,
+		TARGET, MARKED, LAUNCHING, DARING, SECRETIVE, RAMMING
```

Comment:
You should split this into one personality per line. It'll make PR reviews of new personalities easier, since the only line being changed just has the one personality that is added.

Reply:
My next suggested change (see below) will do this for you.